linkerd/proxy/http/src/normalize_uri.rs

```diff
@@ -32,6 +32,22 @@ pub fn layer<M>() -> impl tower::layer::Layer<M, Service = MakeNormalizeUri<M>>
 
 // === impl MakeNormalizeUri ===
 
+impl<T, M> NewService<T> for MakeNormalizeUri<M>
+where
+    T: ShouldNormalizeUri,
+    M: NewService<T>,
+{
+    type Service = NormalizeUri<M::Service>;
+
+    fn new_service(&self, target: T) -> Self::Service {
+        let authority = target.should_normalize_uri();
+        tracing::trace!(?authority, "new");
```

Comment:
nit: here, we use `fmt::Debug` to record the authority, but in `NormalizeUri::call`, we record it with `fmt::Display`: https://github.com/linkerd/linkerd2-proxy/blob/c58ffc9ab43ab5ad05526180c6e82529fe37adb2/linkerd/proxy/http/src/normalize_uri.rs#L118

we should probably pick one and stick with it, to make this more consistent (so you can grep an authority in the logs, etc)?

Reply:
This probably should go away anyway, as it's all on the trace context already